conan/tools/cmake/cmake.py

```diff
@@ -57,38 +58,48 @@ def __init__(self, conanfile):
 
         self._generator = configure_preset["generator"]
         self._toolchain_file = configure_preset.get("toolchainFile")
-        self._cache_variables = configure_preset["cacheVariables"]
-
+        self._presets_cache_variables = configure_preset["cacheVariables"]
         self._cmake_program = "cmake"  # Path to CMake should be handled by environment
 
+    @property
+    def _cache_variables(self):
+        """
+        Get all the variables that should be passed the CLI as ``cmake -Dxxxxx -Dyyyyy ....``
+        """
+        cache_variables = {}
+        generator_folder = self._conanfile.generators_folder
+        if self._toolchain_file:
+            if os.path.isabs(self._toolchain_file):
+                toolchain_file = self._toolchain_file
+            else:
+                toolchain_file = os.path.join(generator_folder, self._toolchain_file)
+            cache_variables["CMAKE_TOOLCHAIN_FILE"] = toolchain_file.replace("\\", "/")
+
+        if self._conanfile.package_folder:
+            pkg_folder = self._conanfile.package_folder.replace("\\", "/")
+            cache_variables["CMAKE_INSTALL_PREFIX"] = pkg_folder
+
+        cache_variables.update(self._presets_cache_variables)
+        return cache_variables
+
     def configure(self, variables=None, build_script_folder=None):
         cmakelist_folder = self._conanfile.source_folder
         if build_script_folder:
             cmakelist_folder = os.path.join(self._conanfile.source_folder, build_script_folder)
 
         build_folder = self._conanfile.build_folder
-        generator_folder = self._conanfile.generators_folder
-
         mkdir(self._conanfile, build_folder)
 
         arg_list = [self._cmake_program]
+        # Running the initial cache file
+        cache_variables = self._cache_variables
+        cache_variables.update(variables or {})
+        update_cmake_initial_cache(self._conanfile, cache_variables)
```

Comment:
I don't like to update the initial cache from here, I would keep calling cmake with the "-D" like before.